### PR TITLE
Add Application backend validation + UI Improvements

### PR DESCRIPTION
**Pull request by dhaselhan**
* UI Improvements to Frontend
* Add new validator service and unit tests
* Add invalid format errors to front end

### CODE REVIEW

File: services/apps/portal/src/alcs/code/code.controller.spec.ts [hunk 1]
```diff
@@ -44,7 +44,6 @@ describe('CodeController', () => {
         uuid: 'fake-uuid',
         name: 'fake-name',
         isFirstNation: false,
-        isActive: true,
```
Comment (mhuseinov) [inline services/apps/portal/src/alcs/code/code.controller.spec.ts:47]: Is it not relevant anymore?

File: services/apps/portal/src/application-review/application-review.controller.ts [hunk 2]
```diff
@@ -156,9 +158,21 @@ export class ApplicationReviewController {
       userLocalGovernment.isFirstNation,
     );
 
+    const validationResult =
+      await this.applicationValidatorService.validateApplication(application);
+
+    if (!validationResult.application) {
+      throw new BaseServiceException(
+        `Invalid application found during LG Submission ${application.fileNumber}`,
```
Comment (mhuseinov) [inline services/apps/portal/src/application-review/application-review.controller.ts:166]: Do I get it right that later this message will be dynamic based on the application type?

**Review by mhuseinov — APPROVED** [review]

File: services/apps/portal/src/alcs/code/code.controller.spec.ts (hunk 1 above)
Comment (dhaselhan) [reply services/apps/portal/src/alcs/code/code.controller.spec.ts:47]: This was a mistake by me, it was never actually sent over from ALCS, so it was always undefined.

File: services/apps/portal/src/application-review/application-review.controller.ts (hunk 2 above)
Comment (dhaselhan) [reply services/apps/portal/src/application-review/application-review.controller.ts:166]: This is more a safeguard, this error should never happen as the application was validated when the Applicant submitted it. So unless something went wrong then the application should still be valid after the LG review is added.

**Pull request merged**